File: streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.17/src/main/scala/org/apache/streampark/flink/core/StreamTableContext.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.flink.core

import org.apache.streampark.common.util.Implicits.JavaList

import org.apache.flink.api.java.utils.ParameterTool
import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
import org.apache.flink.table.api._
import org.apache.flink.table.api.bridge.scala.{StreamStatementSet, StreamTableEnvironment}
import org.apache.flink.table.connector.ChangelogMode
import org.apache.flink.table.module.ModuleEntry
import org.apache.flink.table.resource.ResourceUri
import org.apache.flink.table.types.AbstractDataType
import org.apache.flink.types.Row

class StreamTableContext(
    override val parameter: ParameterTool,
    private val streamEnv: StreamExecutionEnvironment,
    private val tableEnv: StreamTableEnvironment)
  extends FlinkStreamTableTrait(parameter, streamEnv, tableEnv) {

  def this(args: (ParameterTool, StreamExecutionEnvironment, StreamTableEnvironment)) =
    this(args._1, args._2, args._3)

  def this(args: StreamTableEnvConfig) =
    this(FlinkTableInitializer.initialize(args))

  override def fromDataStream[T](dataStream: DataStream[T], schema: Schema): Table =
    tableEnv.fromDataStream[T](dataStream, schema)

  override def fromChangelogStream(dataStream: DataStream[Row]): Table =
    tableEnv.fromChangelogStream(dataStream)

  override def fromChangelogStream(dataStream: DataStream[Row], schema: Schema): Table =
    tableEnv.fromChangelogStream(dataStream, schema)

  override def fromChangelogStream(
      dataStream: DataStream[Row],
      schema: Schema,
      changelogMode: ChangelogMode): Table =
    tableEnv.fromChangelogStream(dataStream, schema, changelogMode)

  override def createTemporaryView[T](
      path: String,
      dataStream: DataStream[T],
      schema: Schema): Unit =
    tableEnv.createTemporaryView[T](path, dataStream, schema)

  override def toDataStream(table: Table): DataStream[Row] = {
    isConvertedToDataStream = true
    tableEnv.toDataStream(table)
  }

  override def toDataStream[T](table: Table, targetClass: Class[T]): DataStream[T] = {
    isConvertedToDataStream = true
    tableEnv.toDataStream[T](table, targetClass)
  }

  override def toDataStream[T](table: Table, targetDataType: AbstractDataType[_]): DataStream[T] = {
    isConvertedToDataStream = true
    tableEnv.toDataStream[T](table, targetDataType)
  }

  override def toChangelogStream(table: Table): DataStream[Row] = {
    isConvertedToDataStream = true
    tableEnv.toChangelogStream(table)
  }

  override def toChangelogStream(table: Table, targetSchema: Schema): DataStream[Row] = {
    isConvertedToDataStream = true
    tableEnv.toChangelogStream(table, targetSchema)
  }

  override def toChangelogStream(
      table: Table,
      targetSchema: Schema,
      changelogMode: ChangelogMode): DataStream[Row] = {
    isConvertedToDataStream = true
    tableEnv.toChangelogStream(table, targetSchema, changelogMode)
  }

  override def createStatementSet(): StreamStatementSet =
    tableEnv.createStatementSet()

  override def useModules(strings: String*): Unit =
    tableEnv.useModules(strings: _*)

  override def createTemporaryTable(path: String, descriptor: TableDescriptor): Unit =
    tableEnv.createTemporaryTable(path, descriptor)

  override def createTable(path: String, descriptor: TableDescriptor): Unit =
    tableEnv.createTable(path, descriptor)

  override def from(descriptor: TableDescriptor): Table =
    tableEnv.from(descriptor)

  override def listFullModules(): Array[ModuleEntry] =
    tableEnv.listFullModules()

  /** @since 1.15 */
  override def listTables(s: String, s1: String): Array[String] =
    tableEnv.listTables(s, s1)

  /** @since 1.15 */
  override def loadPlan(planReference: PlanReference): CompiledPlan =
    tableEnv.loadPlan(planReference)

  /** @since 1.15 */
  override def compilePlanSql(s: String): CompiledPlan =
    tableEnv.compilePlanSql(s)

  /** @since 1.17 */
  override def createFunction(
      path: String,
      className: String,
      resourceUris: JavaList[ResourceUri]): Unit =
    tableEnv.createFunction(path, className, resourceUris)

  /** @since 1.17 */
  override def createFunction(
      path: String,
      className: String,
      resourceUris: JavaList[ResourceUri],
      ignoreIfExists: Boolean): Unit =
    tableEnv.createFunction(path, className, resourceUris, ignoreIfExists)

  /** @since 1.17 */
  override def createTemporaryFunction(
      path: String,
      className: String,
      resourceUris: JavaList[ResourceUri]): Unit =
    tableEnv.createTemporaryFunction(path, className, resourceUris)

  /** @since 1.17 */
  override def createTemporarySystemFunction(
      name: String,
      className: String,
      resourceUris: JavaList[ResourceUri]): Unit =
    tableEnv.createTemporarySystemFunction(name, className, resourceUris)

  /** @since 1.17 */
  override def explainSql(
      statement: String,
      format: ExplainFormat,
      extraDetails: ExplainDetail*): String =
    tableEnv.explainSql(statement, format, extraDetails: _*)
}

File: streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.17/src/main/java/org/apache/streampark/flink/core/StreamTableContext.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.flink.core;

import org.apache.flink.api.java.utils.ParameterTool;
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment;
import org.apache.flink.table.api.bridge.scala.StreamTableEnvironment;

import scala.Tuple3;

/**
 * Integration api of stream and table
 */
public class StreamTableContext extends FlinkStreamTableTrait {

    private final StreamExecutionEnvironment streamEnv;
    private final StreamTableEnvironment tableEnv;

    public StreamTableContext(
                              ParameterTool parameter,
                              StreamExecutionEnvironment streamEnv,
                              StreamTableEnvironment tableEnv) {
        super(parameter, streamEnv, tableEnv);
        this.streamEnv = streamEnv;
        this.tableEnv = tableEnv;
    }

    public StreamTableContext(
                              Tuple3<ParameterTool, StreamExecutionEnvironment, StreamTableEnvironment> args) {
        this(args._1(), args._2(), args._3());
    }

    public StreamTableContext(StreamTableEnvConfig args) {
        this(FlinkTableInitializer.initialize(args));
    }

    @Override
    public <T> org.apache.flink.table.api.Table fromDataStream(
                                                               org.apache.flink.streaming.api.scala.DataStream<T> dataStream,
                                                               org.apache.flink.table.api.Schema schema) {
        return tableEnv.fromDataStream(dataStream, schema);
    }

    @Override
    public org.apache.flink.table.api.Table fromChangelogStream(
                                                                org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> dataStream) {
        return tableEnv.fromChangelogStream(dataStream);
    }

    @Override
    public org.apache.flink.table.api.Table fromChangelogStream(
                                                                org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> dataStream,
                                                                org.apache.flink.table.api.Schema schema) {
        return tableEnv.fromChangelogStream(dataStream, schema);
    }

    @Override
    public org.apache.flink.table.api.Table fromChangelogStream(
                                                                org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> dataStream,
                                                                org.apache.flink.table.api.Schema schema,
                                                                org.apache.flink.table.connector.ChangelogMode changelogMode) {
        return tableEnv.fromChangelogStream(dataStream, schema, changelogMode);
    }

    @Override
    public <T> void createTemporaryView(
                                        String path,
                                        org.apache.flink.streaming.api.scala.DataStream<T> dataStream,
                                        org.apache.flink.table.api.Schema schema) {
        tableEnv.createTemporaryView(path, dataStream, schema);
    }

    @Override
    public org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> toDataStream(
                                                                                                    org.apache.flink.table.api.Table table) {
        isConvertedToDataStream = true;
        return tableEnv.toDataStream(table);
    }

    @Override
    public <T> org.apache.flink.streaming.api.scala.DataStream<T> toDataStream(
                                                                               org.apache.flink.table.api.Table table,
                                                                               Class<T> targetClass) {
        isConvertedToDataStream = true;
        return tableEnv.toDataStream(table, targetClass);
    }

    @Override
    public <T> org.apache.flink.streaming.api.scala.DataStream<T> toDataStream(
                                                                               org.apache.flink.table.api.Table table,
                                                                               org.apache.flink.table.types.AbstractDataType<?> targetDataType) {
        isConvertedToDataStream = true;
        return tableEnv.toDataStream(table, targetDataType);
    }

    @Override
    public org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> toChangelogStream(
                                                                                                         org.apache.flink.table.api.Table table) {
        isConvertedToDataStream = true;
        return tableEnv.toChangelogStream(table);
    }

    @Override
    public org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> toChangelogStream(
                                                                                                         org.apache.flink.table.api.Table table,
                                                                                                         org.apache.flink.table.api.Schema targetSchema) {
        isConvertedToDataStream = true;
        return tableEnv.toChangelogStream(table, targetSchema);
    }

    @Override
    public org.apache.flink.streaming.api.scala.DataStream<org.apache.flink.types.Row> toChangelogStream(
                                                                                                         org.apache.flink.table.api.Table table,
                                                                                                         org.apache.flink.table.api.Schema targetSchema,
                                                                                                         org.apache.flink.table.connector.ChangelogMode changelogMode) {
        isConvertedToDataStream = true;
        return tableEnv.toChangelogStream(table, targetSchema, changelogMode);
    }

    @Override
    public void useModules(String... strings) {
        tableEnv.useModules(strings);
    }

    @Override
    public org.apache.flink.table.module.ModuleEntry[] listFullModules() {
        return tableEnv.listFullModules();
    }

    @Override
    public org.apache.flink.table.api.bridge.scala.StreamStatementSet createStatementSet() {
        return tableEnv.createStatementSet();
    }

    @Override
    public void createTemporaryTable(String path, org.apache.flink.table.api.TableDescriptor descriptor) {
        tableEnv.createTemporaryTable(path, descriptor);
    }

    @Override
    public void createTable(String path, org.apache.flink.table.api.TableDescriptor descriptor) {
        tableEnv.createTable(path, descriptor);
    }

    @Override
    public org.apache.flink.table.api.Table from(org.apache.flink.table.api.TableDescriptor descriptor) {
        return tableEnv.from(descriptor);
    }

    /** @since 1.15 */
    @Override
    public String[] listTables(String s, String s1) {
        return tableEnv.listTables(s, s1);
    }

    /** @since 1.15 */
    @Override
    public org.apache.flink.table.api.CompiledPlan loadPlan(org.apache.flink.table.api.PlanReference planReference) {
        return tableEnv.loadPlan(planReference);
    }

    /** @since 1.15 */
    @Override
    public org.apache.flink.table.api.CompiledPlan compilePlanSql(String s) {
        return tableEnv.compilePlanSql(s);
    }

    /** @since 1.17 */
    @Override
    public void createFunction(
                               String path, String className,
                               java.util.List<org.apache.flink.table.resource.ResourceUri> resourceUris) {
        tableEnv.createFunction(path, className, resourceUris);
    }

    /** @since 1.17 */
    @Override
    public void createFunction(
                               String path,
                               String className,
                               java.util.List<org.apache.flink.table.resource.ResourceUri> resourceUris,
                               boolean ignoreIfExists) {
        tableEnv.createFunction(path, className, resourceUris, ignoreIfExists);
    }

    /** @since 1.17 */
    @Override
    public void createTemporaryFunction(
                                        String path, String className,
                                        java.util.List<org.apache.flink.table.resource.ResourceUri> resourceUris) {
        tableEnv.createTemporaryFunction(path, className, resourceUris);
    }

    /** @since 1.17 */
    @Override
    public void createTemporarySystemFunction(
                                              String name, String className,
                                              java.util.List<org.apache.flink.table.resource.ResourceUri> resourceUris) {
        tableEnv.createTemporarySystemFunction(name, className, resourceUris);
    }

    /** @since 1.17 */
    @Override
    public String explainSql(
                             String statement,
                             org.apache.flink.table.api.ExplainFormat format,
                             org.apache.flink.table.api.ExplainDetail... extraDetails) {
        return tableEnv.explainSql(statement, format, extraDetails);
    }

}
